source/common/http/http1/codec_impl.cc

```diff
@@ -101,9 +101,12 @@ void StreamEncoderImpl::encodeHeaders(const HeaderMap& headers, bool end_stream)
                    Headers::get().TransferEncoding.get().size(),
                    Headers::get().TransferEncodingValues.Chunked.c_str(),
                    Headers::get().TransferEncodingValues.Chunked.size());
+      // We do not aply chunk encoding for HTTP upgrades. Any incoming chunks
+      // will be through-proxied by maybeDirectDispatch untouched.
```

Comment:
Isn't not chunk encoding the stream orthogonal from how decoding happens on the other end? I think maybe the comment just needs clarification that in the upgrade case we don't chunk encode because we are basically dealing with "raw" bytes on the decode side? (Rephrase however you want just trying to make sure I and future readers understand.)